core/src/main/java/org/jivesoftware/sparkimpl/profile/AvatarPanel.java

```diff
@@ -291,7 +293,49 @@ public void setParentDialog(Dialog dialog) {
         this.dlg = dialog;
     }
 
+    /*
+     * Crop the given image file to the target dimensions 
+     * without changing aspect ratio.
+     * Save the resultant file into disk and return a reference
+     * to it.  
+     */
+    private File cropImageFile(File selectedFile){
+
+        int targetImageWidth = 100;
+        int targetImageHeight = 100;
+        File imageFile = null;
+
+        try {
+            BufferedImage avatarImageBuffered = ImageIO.read(selectedFile);
+            BufferedImage croppedImage;
+
+            int currentImageWidth = avatarImageBuffered.getWidth(null);
+            int currentImageHeight = avatarImageBuffered.getHeight(null);
+            
+            if(currentImageHeight == targetImageHeight && currentImageWidth == targetImageHeight){
+            	//no need to crop
+                croppedImage = avatarImageBuffered;
+                imageFile = selectedFile;
+            }else{
+                double targetImageRatio = (double)targetImageWidth / (double)targetImageHeight;
+
+                if (currentImageWidth < targetImageRatio * currentImageHeight) {
+                    currentImageHeight = new Double((new Double(currentImageWidth )/ new Double(targetImageRatio))).intValue();
+                } else {
+                    currentImageWidth = new Double(((double)currentImageHeight * targetImageRatio)).intValue();
+                }
+                imageFile = new File(SparkManager.getUserDirectory(), "personal.png");
```

Comment:
I severely dislike creating a new file, without asking for permission to the user. We've got a couple of options, all of which I think are better than creating a random file in the user directory:

- Don't create an explicit file, but process the original image each time that we need the smaller file (this could be CPU intensive)
- Buffer the outcome of processing (potentially storing them in memory for a while). This could be memory intensive
- If you must, store the small file in a temporary file.